src/community/modularity.c

```diff
@@ -216,6 +216,46 @@ int igraph_modularity(const igraph_t *graph,
     return IGRAPH_SUCCESS;
 }
 
+static int igraph_i_modularity_matrix_get_adjacency(
+           const igraph_t *graph, igraph_matrix_t *res,
+           const igraph_vector_t *weights, igraph_bool_t directed) {
+    /* Specifically used to handle weights and/or ignore direction */
+    igraph_eit_t edgeit;
+    long int no_of_nodes = igraph_vcount(graph);
+    igraph_integer_t from, to;
+
+    IGRAPH_CHECK(igraph_matrix_resize(res, no_of_nodes, no_of_nodes));
+    igraph_matrix_null(res);
+    IGRAPH_CHECK(igraph_eit_create(graph, igraph_ess_all(0), &edgeit));
+    IGRAPH_FINALLY(igraph_eit_destroy, &edgeit);
+
+    if (weights) {
+        while (!IGRAPH_EIT_END(edgeit)) {
+            igraph_integer_t edge = IGRAPH_EIT_GET(edgeit);
+            igraph_edge(graph, edge, &from, &to);
```

Comment:
Let's use the newer `IGRAPH_FROM` and `IGRAPH_TO`. These are macros, which means that they will always be inlined. `igraph_edge` will cause a function call (unless link-time optimization is used and the compiler can inline it) which is potentially slow.

While performance is likely not critical here, I'd prefer `IGRAPH_FROM` and `IGRAPH_TO` in new code.